### PR TITLE
feat: add acct_params_get opcode

### PR DESCRIPTION
**Pull request by vuvth**
## Proposed Changes

+ Add AcctParamsGet to handle acct_params_get opcode(check opcode-list.ts)
+ Add acct_params_get to parser (parser.ts).
+ Create `create` for interpreter.getAcccount (check interpreter.ts). When `create` flag is false we will throws exception if account is not found else if `create` is true we will create new account and add this account to context.
+ Unit test for them

## TODO
- [x] Update changelog.md after PR approval.

Please review my PR! Thank you a lot.

### CODE REVIEW

File: packages/runtime/src/errors/errors-list.ts [hunk 1]
```diff
@@ -377,6 +377,13 @@ maximun uint128`,
 		title: "Execution mode not valid",
 		description: `Execution mode not valid`,
 	},
+	UNKNOWN_ACCT_FIELD: {
+		number: 1053,
```
Comment (robert-zaremba) [inline packages/runtime/src/errors/errors-list.ts:381]:
```suggestion
		number: 1055,
```

File: packages/runtime/src/interpreter/interpreter.ts [hunk 2]
```diff
@@ -107,14 +108,28 @@ export class Interpreter {
...
+	
 	/**
 	 * Beginning from TEALv4, user can directly pass address instead of index to Txn.Accounts.
 	 * However, the address must still be present in tx.Accounts OR should be equal to Txn.Sender
 	 * @param accountPk public key of account
 	 * @param line line number in TEAL file
 	 * https://developer.algorand.org/articles/introducing-algorand-virtual-machine-avm-09-release/
 	 */
-	private _getAccountFromAddr(accountPk: Uint8Array, line: number): AccountStoreI {
+	private _getAccountFromAddr(accountPk: Uint8Array, line: number, strict = true): AccountStoreI {
```
Comment (robert-zaremba) [inline packages/runtime/src/interpreter/interpreter.ts:132]: need to add docs about `strict`

Comment (robert-zaremba) [reply packages/runtime/src/interpreter/interpreter.ts:132]: Also, I suggest to rename `strict` to `create`

File: packages/runtime/src/interpreter/interpreter.ts [hunk 3]
```diff
@@ -145,7 +160,13 @@ export class Interpreter {
 			compareArray(accountPk, decodeAddress(getApplicationAddress(appID)).publicKey)
 		) {
 			const address = encodeAddress(pkBuffer);
-			const account = this.runtime.ctx.state.accounts.get(address);
+			let account = this.runtime.ctx.state.accounts.get(address);
+			// strict is turn on we will create account store
+			// we will create account if it not exist
+			if (!strict) {
+				account = this.createAccountIfAbsent(address);
+			}
```
Comment (robert-zaremba) [inline packages/runtime/src/interpreter/interpreter.ts:168]:
```suggestion
			let account = create ? this.createAccountIfAbsent(address) : this.runtime.ctx.state.accounts.get(address);
```

File: packages/runtime/src/interpreter/interpreter.ts [hunk 4]
```diff
@@ -181,9 +204,12 @@ export class Interpreter {
 				}
 				address = encodeAddress(pkBuffer);
 				account = this.runtime.ctx.state.accounts.get(address);
+				if (!strict) {
+					account = this.createAccountIfAbsent(address);
+				}
```
Comment (robert-zaremba) [inline packages/runtime/src/interpreter/interpreter.ts:209]:
```suggestion
				account = create ? this.createAccountIfAbsent(address) : this.runtime.ctx.state.accounts.get(address);
```

File: packages/runtime/src/interpreter/interpreter.ts [hunk 5]
```diff
@@ -158,12 +179,14 @@ export class Interpreter {
...
-	 * Throws exception if account is not found.
+	 * when `strict` is true we will throws exception if account is not found. 
+	 * when `strict` is false we will create new account and add it to context.
 	 * @param accountRef index of account to fetch from account list
 	 * @param line line number
+	 * @param strict strict flag
 	 * NOTE: index 0 represents txn sender account
 	 */
-	getAccount(accountRef: StackElem, line: number): AccountStoreI {
+	getAccount(accountRef: StackElem, line: number, strict = true): AccountStoreI {
```
Comment (robert-zaremba) [inline packages/runtime/src/interpreter/interpreter.ts:189]: let's rename strict as well

File: packages/runtime/src/interpreter/opcode-list.ts [hunk 6]
```diff
@@ -4562,3 +4563,70 @@ export class AppParamsGet extends Op {
...
 	}
 }
+
+export class AcctParamsGet extends Op {
+	readonly interpreter: Interpreter;
+	readonly line: number;
+	readonly field: string;
+	/**
+	 * Asserts 1 arguments are passed.
+	 * @param args Expected arguments: [] // none
```
Comment (robert-zaremba) [inline packages/runtime/src/interpreter/opcode-list.ts:4573]:
```suggestion
	 * @param args Expected arguments: [account_param]
```

File: packages/runtime/src/interpreter/opcode-list.ts [hunk 7]
```diff
@@ -4562,3 +4563,70 @@ export class AppParamsGet extends Op {
...
+				break;
+			}
+			case "AcctAuthAddr": {
+				if (accountInfo.getSpendAddress() === accountInfo.address) {
+					value = ZERO_ADDRESS;
+				} else {
+					value = Buffer.from(decodeAddress(accountInfo.getSpendAddress()).publicKey);
+				}
+				break;
+			}
```
Comment (robert-zaremba) [inline packages/runtime/src/interpreter/opcode-list.ts:4622]: add a default case, which will return an error , something like `'unknown account parameter "%s"'`

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 8]
```diff
@@ -6252,4 +6254,118 @@ describe("Teal Opcodes", function () {
...
+		let interpreter: Interpreter;
+		let alice: AccountStoreI;
+		let bob: AccountStoreI;
+		let op: AcctParamsGet;
+		const zeroBalanceAddr = "WWYNX3TKQYVEREVSW6QQP3SXSFOCE3SKUSEIVJ7YAGUPEACNI5UGI4DZCE";
+
+		this.beforeEach(() => {
+			interpreter = new Interpreter();
+			interpreter.runtime = new Runtime([]);
+			[alice, bob] = interpreter.runtime.defaultAccounts()
```
Comment (robert-zaremba) [inline packages/runtime/test/src/interpreter/opcode-list.ts:6270]: we can use default accounts :) cc: @sczembor

File: packages/runtime/test/src/interpreter/opcode-list.ts [hunk 9]
```diff
@@ -6252,4 +6254,118 @@ describe("Teal Opcodes", function () {
...
+
+			// valid address but not in tx accounts list
+			stack.push(parsing.stringToBytes("01234567890123456789012345678901"));
+			expectRuntimeError(
+				() => op.execute(stack),
+				RUNTIME_ERRORS.TEAL.ADDR_NOT_FOUND_IN_TXN_ACCOUNT
+			)
+		});
+
+		it("Should throw error if top element in stack is not address type", () => {
```
Comment (robert-zaremba) [inline packages/runtime/test/src/interpreter/opcode-list.ts:6361]:
```suggestion
		it("Should throw error if top element in stack is not an address", () => {
```

**Review by robert-zaremba — APPROVED** [review]
pre approving. Left few suggestions.

File: packages/runtime/src/interpreter/opcode-list.ts (hunk 7 above)
Comment (vuvth) [reply packages/runtime/src/interpreter/opcode-list.ts:4622]: We can't create Opcode class if the parameter invalid(will throw error) so we don't need handle this case.

**Pull request merged**